camel/memories/blocks/graphdb_block.py

```diff
@@ -0,0 +1,224 @@
+        for record in records:
+            properties = record.to_dict()
+
+            def flatten_and_convert(d):
+                for key, value in list(d.items()):
+                    if isinstance(value, Enum):
+                        d[key] = value.name  # Convert Enum to its name
+                    elif value is None:
+                        d[key] = (
+                            '__none__'  # Convert None to a sentinel string
+                        )
+                    elif isinstance(value, dict):
+                        if value:  # If dictionary is not empty
+                            for sub_key, sub_value in flatten_and_convert(
+                                value
+                            ).items():
+                                d[f"{key}_{sub_key}"] = sub_value
+                            del d[key]
+                        else:
+                            d[key] = '__empty_dict__'
+                    elif isinstance(value, list):
+                        d[key] = [
+                            str(i)
+                            if not isinstance(i, (str, int, float, bool))
+                            else i
+                            for i in value
+                        ]
+                    else:
+                        d[key] = (
+                            str(value)
+                            if not isinstance(value, (str, int, float, bool))
+                            else value
+                        )
+                return d
+
+            properties = flatten_and_convert(properties)
+
+            content = record.message.content
+
+            if not isinstance(content, str):
+                logger.error(
+                    "Expected content to be a string, got"
+                    f" {type(content)}: {content}"
+                )
+                continue
+
+            embedding = self.embedding.embed(content)
+            properties['embedding'] = embedding
+
+            try:
+                self.storage.add_node(label="Message", properties=properties)
+                logger.info("Node with label 'Message' added successfully.")
+            except Exception as e:
+                logger.error(f"Failed to add node with label 'Message': {e!s}")
+
+    def delete_records(self, records: List[MemoryRecord]) -> None:
+        """Deletes records to the graph database."""
+        for record in records:
+            content = record.message.content
+
+            if not isinstance(content, str):
+                logger.error(
+                    "Expected content to be a string,"
+                    f"got {type(content)}: {content}"
+                )
+                continue
+
+            try:
+                # Assume that record deletion involves removing
+                # nodes and relationships
+                self.storage.delete_triplet(
+                    subj=content, obj=content, rel="NEXT"
+                )
```

Comment:
same as above, use entity from message content rather than content it self would be better? BTW why `rel="NEXT"` here, how it used?